hyperspy/signal.py

```diff
@@ -4492,12 +4494,22 @@ def map(
         >>> im.map(scipy.ndimage.gaussian_filter, sigma=sigmas)
 
         """
-        # Sepate ndkwargs
-        ndkwargs = ()
-        for key, value in list(kwargs.items()):
-            if isinstance(value, BaseSignal):
-                ndkwargs += ((key, value),)
-
+        if self.axes_manager.navigation_shape == () and self._lazy:
+            print("Converting signal to a non-lazy signal because there are no nav dimensions")
+            self.compute()
+        # Sepate ndkwargs depending on if they are BaseSignals.
```

Comment:
It should be noted that this actually works with _all_ signals which inherits `BaseSignal`. Thus, the current implementation works fine with `Signal2D` and `Signal1D` as well.